### PR TITLE
Transfer Fee fix

### PR DESCRIPTION
**Pull request by JosephAMumford**
I simplified the transaction function to calculate only what is needed, from how I understand the charge/payment process from the codesy website.  This may affect other source files that call the transaction amounts function since I deleted a few variables, which seemed to exist for testing purposes.  

The important change is that the final payout amount is determined by the initial payout divided by (1 + transfer_fee).  Transfer fee is then initial payout - final payout. #449

### CODE REVIEW

**Comment by aprilchomp** [comment]
Thank you for your contribution @JosephAMumford! Your code will be reviewed ASAP!

**Comment by groovecoder** [comment]
Travis CI did the first review and found some issues ...

https://travis-ci.org/codesy/codesy/jobs/289764327 looks like it caught a bunch of white-space/indentation issues.

https://travis-ci.org/codesy/codesy/jobs/289764326 shows a problem multiplying a `Decimal` type with a `float` type.

```
  File "/home/travis/build/codesy/codesy/payments/utils.py", line 36, in transaction_amounts

    charge_amount = amount + codesy_fee_amount + (stripe_fee_amount * 0.5)

TypeError: unsupported operand type(s) for *: 'Decimal' and 'float'
```

@jdungan has more experience with the right way to handle `Decimal` with `float` I think.

File: payments/utils.py [hunk 1]
```diff
@@ -16,92 +16,51 @@
...
-
-    return {
-        'amount': amount,
-        'charge_amount': round_penny(charge_amount),
-        'payout_amount': payout_amount,
+	codesy_fee_amount = calculate_codesy_fee(amount)
+	stripe_fee_amount = calculate_stripe_fee(amount + codesy_fee_amount)
+	
+	# this is the total to get charged to codesy's account
+	charge_amount = amount + codesy_fee_amount + (stripe_fee_amount * 0.5)
```
Comment (groovecoder) [inline payments/utils.py]: This is causing
```
  File "/home/travis/build/codesy/codesy/payments/utils.py", line 36, in transaction_amounts

    charge_amount = amount + codesy_fee_amount + (stripe_fee_amount * 0.5)

TypeError: unsupported operand type(s) for *: 'Decimal' and 'float'
```

**Comment by groovecoder** [comment]
Should I assume you're working with @jdungan on this right now? Still planning to work on it?

**Comment by JosephAMumford** [comment]
I didn’t realize I closed, must have hit that button by mistake.  My original code was creating a bunch of errors when it was called in other parts of the program so I just changed the transfer fee function.  I’m not sure if @jdungan has reviewed it and though my formula works for that part as I understand the fee process, there could still be an issue.

**Comment by jdungan** [comment]
@JosephAMumford Sorry I haven't had a chance to test.   You will need to included the overage value and write a test that proves it is always zero.

**Comment by groovecoder** [comment]
Latest commit looks good. 👍 for adding a test that verifies this new/updated logic.

**Comment by groovecoder** [comment]
Should add this to `payments/tests/util_tests.py` at the end of both blocks in the `test_fixed_amounts` function:
```
             self.assertEqual(values['transfer_overage'], Decimal('0.00'))
```

Can run the test like so:
```
python manage.py test payments.tests.util_tests
```

Then you can mess with the `utils.transaction_amounts` logic until the test is passing each time. Get @jdungan to help with that. :)

**Comment by groovecoder** [comment]
An option is to just give the overage to the payout. So ...

```
diff --git a/payments/utils.py b/payments/utils.py
index ad4b178..04296e6 100644
--- a/payments/utils.py
+++ b/payments/utils.py
@@ -91,6 +91,10 @@ def transaction_amounts(amount):
         - actual_transfer_fee
     )

+    if (transfer_overage >= 0):
+        payout_amount += transfer_overage
+        transfer_overage = 0
+
     return {
         'amount': amount,
         'charge_amount': round_penny(charge_amount),
```

**Comment by jdungan** [comment]
Adding the test assertion suggested above identifies a payout amount of 18 still gets an overage of $0.01.  You can confirm this using the management command: `python manage.py calculate 18`
